js/Thread.js

```diff
@@ -1,28 +1,30 @@
-import {
-  NativeModules,
-  DeviceEventEmitter,
-} from 'react-native';
+import { NativeModules, DeviceEventEmitter } from "react-native";
 
 const { ThreadManager } = NativeModules;
 
 export default class Thread {
   constructor(jsPath) {
-    if (!jsPath || !jsPath.endsWith('.js')) {
-      throw new Error('Invalid path for thread. Only js files are supported');
+    if (!jsPath || !jsPath.endsWith(".js")) {
+      throw new Error("Invalid path for thread. Only js files are supported");
     }
 
     this.id = ThreadManager.startThread(jsPath.replace(".js", ""))
-      .then(id => {
-        DeviceEventEmitter.addListener(`Thread${id}`, (message) => {
-          !!message && this.onmessage && this.onmessage(message);
+      .then((id) => {
+        DeviceEventEmitter.addListener(`Thread`, (msg) => {
+          if (!msg) return;
+          const { message, id: threadId } = JSON.parse(msg);
+          if (id === threadId && this.onmessage)
+            this.onmessage(JSON.stringify(message));
         });
         return id;
       })
-      .catch(err => { throw new Error(err) });
+      .catch((err) => {
+        throw new Error(err);
+      });
```

Comment:
does this catch do anything?
```suggestion
```

Reply:
This is existing code from the lib, but It seems it does nothing :)